zap/src/main/java/org/zaproxy/zap/extension/alert/AlertAPI.java

```diff
@@ -442,15 +645,34 @@ private int getRiskId(JSONObject parameters) throws ApiException {
         try {
             riskId = Integer.parseInt(riskIdParam);
         } catch (NumberFormatException e) {
-            throwInvalidRiskId();
```

Comment:
It's better to keep this method to avoid the duplication.

Reply:
What duplication? Both `getRiskId` and the new `getConfidenceId` are going to use `throwIllegalApiParameter` doesn't that reduce duplication?

Reply:
It's duplicating the message (`Constant.messages...`).

Reply:
But that doesn't go away, it just means I need to add a throwInvalidConfidenceId and still have two message keys.... unless we go back to non-i18n messages but I thought we agreed to that in the other comments?

Reply:
I've reduced the message entries from 6 to 2 in the latest iteration, maybe that addresses your concern?

Reply:
Spoke in IRC, I think we're on the same page now.